apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsLinkControl.tsx

```diff
@@ -0,0 +1,117 @@
+import type { ControlProps, RankedTester } from "@jsonforms/core"
+import {
+  Box,
+  FormControl,
+  HStack,
+  Icon,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Text,
+} from "@chakra-ui/react"
+import { and, isStringControl, rankWith, schemaMatches } from "@jsonforms/core"
+import { withJsonFormsControlProps } from "@jsonforms/react"
+import { FormLabel, Input, Tab } from "@opengovsg/design-system-react"
+import { BiEnvelopeOpen, BiFile, BiFileBlank, BiLink } from "react-icons/bi"
+
+import { JSON_FORMS_RANKING } from "~/constants/formBuilder"
+
+export const jsonFormsLinkControlTester: RankedTester = rankWith(
+  JSON_FORMS_RANKING.LinkControl,
+  and(
+    isStringControl,
+    schemaMatches((schema) => schema.format === "link"),
+  ),
+)
+
+export function JsonFormsLinkControl({
+  data,
+  label,
+  handleChange,
+  path,
+  description,
+  required,
+}: ControlProps) {
+  const LINK_TYPES = [
```

Comment:
nit: if this is a constant, might want to move it outside of the react component so this is no recreated on every render

Reply:
Moved out in 950c8ed0b863961599692a2057cc26ee605d469c.